src/sentry/integrations/slack/utils/channel.py

```diff
@@ -173,3 +178,115 @@ def get_channel_id_with_timeout(
+        2. channel_id: string or `None`
+        3. timed_out: boolean (whether we hit our self-imposed time limit)
+    """
+
+    payload = {
+        "exclude_archived": False,
+        "exclude_members": True,
+        "types": "public_channel,private_channel",
+    }
+
+    time_to_quit = time.time() + timeout
+
+    client = SlackClient(integration_id=integration.id)
+    id_data: Optional[Tuple[str, Optional[str], bool]] = None
+    found_duplicate = False
+    prefix = ""
+    channel_id = None
+    try:  # Check for channel
+        channel_id = check_for_channel(client, name)
+        prefix = "#"
+    except ApiError as e:
+        if str(e) != "channel_not_found":
+            # return prefix, None, False
+            raise e
+        # Check if user
+        while True:
+            cursor = ""
+            # Slack limits the response of `<list_type>.list` to 1000 channels
+            try:
+                items = client.get("/users.list", params=dict(payload, cursor=cursor, limit=1000))
+            except ApiRateLimitedError as e:
+                logger.info("rule.slack.user_list_rate_limited", extra={"error": str(e)})
+                raise e
+            except ApiError as e:
+                logger.info("rule.slack.user_list_rate_limited", extra={"error": str(e)})
+                return prefix, None, False
+
+            if not isinstance(items, dict):
+                continue
+
+            for c in items["members"]:
+                # The "name" field is unique (this is the username for users)
+                # so we return immediately if we find a match.
+                # convert to lower case since all names in Slack are lowercase
+                if name and c["name"].lower() == name.lower():
+                    prefix = "@"
+                    return prefix, c["id"], False
+                # If we don't get a match on a unique identifier, we look through
+                # the users' display names, and error if there is a repeat.
+                profile = c.get("profile")
+                if profile and profile.get("display_name") == name:
+                    if id_data:
+                        found_duplicate = True
+                    else:
+                        prefix = "@"
+                        id_data = (prefix, c["id"], False)
+
+            cursor = items.get("response_metadata", {}).get("next_cursor", None)
+            if time.time() > time_to_quit:
+                return prefix, None, True
+
+            if not cursor:
+                break
+
+        if found_duplicate:
+            raise DuplicateDisplayNameError(name)
+        elif id_data:
+            return id_data
+
+    return prefix, channel_id, False
+
+
+def check_for_channel(
+    client: SlackClient,
+    name: str,
+) -> str:
+
+    msg_response = client.post(
+        "/chat.scheduleMessage",
+        data={
+            "channel": name,
+            "text": "Sentry looking for channel",
```

Comment:
If for some weird reason the next code block fails to execute, this would send a message to the channel if found, right? Maybe just to be safe let's make it say something like "Sentry is verifying your channel is accessible so that we can send you alert rule notifications". Just on the extreme off-chance someone actually sees it 😬